src/pages/StatusPage.tsx

```diff
@@ -0,0 +1,293 @@
+import { useLocation, useNavigate } from "react-router-dom";
```

Comment:
can we break this page for simplification or it would became hell to main this file it really has a lot of logic and JSX